rundeckapp/grails-app/services/rundeck/services/LogFileStorageService.groovy

```diff
@@ -636,6 +636,11 @@ class LogFileStorageService
             return
         }
         //multi storage available
+        //avoid multiple storage requests for the same Execution
+        def orig = LogFileStorageRequest.findByExecution(e)
+        if(orig){
+            return
+        }
         LogFileStorageRequest request = createStorageRequest(e, '*')
```

Comment:
I believe the flush in `createStorageRequest` would raise an error if the unique index constraint were violated? Might be able to avoid an extra db call.

In either case I believe that because the transaction is started in `ExecutionEventsService` two could being going at the same time and not "see" each other, so the request could be queued and the commit would fail when `ExecutionEventsService.executionComplete` returns. I'm not sure if that's a problem per-say, just an observation.

Reply:
yes, i added that findBy check earlier in my test, it's maybe not useful

Reply:
these are both kind of an experimental fix since we haven't reproduced the conditions that cause the eventual problem